docs/handbook/sunray.md

```diff
@@ -0,0 +1,359 @@
+-       # check runtime configuration via ifconfig.
+-       IPADDR=$(ifconfig $1 | awk '
+-               /[      ]inet /{
+-                       for (i = 1; i < NF; ++i) {
+-                               if ($i == "inet") {
+-                                       print $(i+1)
+-                               }
+-                       }
+-               }')
++       IPADDR=`ipadm show-addr -p -o ADDR $1/ | awk '{split($1,a,"/"); print a[1] }'`
+        if [ -z "$IPADDR" ]; then
+            print -u2 "Error in parsing of ifconfig addresses"
+            exit 2
+```
+
+### JRE
+
+The SunRay Software needs the Sun/Oracle JRE 1.7
+
+## Config
+
+You can run `utconfig` and `utadm` run in common manner
+
+### GDM
+
+the gdm service has to be enabled
+
+```
+# svcadm enable graphical-login/gdm
+```
+
+# Revive GNOME on current Hipster
+
+Be sure you have installed
+
+```shell
+# pkg install libwnck
+# pkg install libbonoboui
+# pkg install gnome-themes-standard
+```
+
+these must not be empty metapackages or obsoleted packages.
+
+## update older Hipster
+
+If you have SunRay running on Hipster with GNOME, you can prevent remove GNOME with freeze the GNOME packages and release the version locks
+
+```shell
+# pkg freeze gdm gnome-session gnome-panel metacity libgnomekbd gnome-settings-daemon gweather
+
+# pkg change-facet facet.version-lock.library/desktop/gnome-desktop=false
+# pkg change-facet facet.version-lock.library/desktop/gnome-menus=false
+# pkg change-facet facet.version-lock.gnome/gnome-session=false
+# pkg change-facet facet.version-lock.system/display-manager/gdm=false
+# pkg change-facet version-lock.library/desktop/libgweather=false
+# pkg change-facet version-lock.gnome/window-manager/metacity=false
+```
+
+## install from current Hipster (2019)
+
+If you install new Hipster it is necessary to have the GNOME packages before (or after) obsoleting
+
+The last know version of OI GNOME packages are:
+
+    libgweather@2.30.3
+    gnome-session@2.32.1
+    gnome-panel@2.32.1
+    metacity@2.30.3
+    gdm@2.30.7
+    libgnomekbd@2.30.0
+    gnome-settings-daemon@2.32.1
+
+These are still in hipster repo, but are obsoleted empty metapackages.
+
+[here](http://pkg.toc.de/sunray/)  can you found the old GNOME packages with newer release date, so that these should be able to install it on current Hipster.
+For that add the publisher in that kind, that it can override packages of publisher openindiana.org
+
+```shell
+# pkg set-publisher --search-before=openindiana.org -g http://pkg.toc.de/sunray sunray
+# pkg set-publisher --non-sticky openindiana.org
+```
+
+and install the all packages from publisher sunray.
+
+
+**The version locks have to release anyway.**
+
+#### Xscreensaver
+
+latest Hipster delivers Xscreensaver only in 64bit. The PAM module of SunRay are only shipped in 32bit.
+So we need a Xscreensaver package with 32bit bins (recompile with enabled 32bit or take it from [here](http://pkg.toc.de/sunray) ).
```

Comment:
The user was asked to install all packages from the sunray publisher, so they already have your 32-bit XScreenSaver. Show the user how to install XScreenSaver from your repo directly (drop the compile yourself note), or drop the XScreenSaver section completely.